File: duorou-ui/mock/studentSpecialty.ts
import { Request, Response } from 'express';
// 代码中会兼容本地 service mock 以及部署站点的静态数据
export default {
  // GET POST 可省略
  'GET /api/student_specialty/list': {
    "code": 200,
    "status": "ok",
    "success": true,
    "message": "success",
    "data": [
      {
        "id": "22",
        "createDate": "2021-06-07 05:37:16",
        "modifyDate": "2021-06-07 05:37:20",
        "studentId": "7",
        "studentName": "7号",
        "userId": "1",
        "specialtyId": "31",
        "specialtyName": "游泳",
        "campusId": 6,
        "termId": 2,
        "rightTime": null,
        "classIds": null,
        "isEnable": true
      }
    ],
    "serverTime": "1623843818861"
  },
  'POST /api/specialty/cancel_subscribe': (req: Request, res: Response) => {
    res.send({ status: 'ok'});
  },
};

File: duorou-backend/src/main/java/com/duorou/backend/model/entity/StudentSpecialty.java
package com.duorou.backend.model.entity;

import com.baomidou.mybatisplus.annotation.TableName;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("w_student_specialty")
public class StudentSpecialty extends BaseInputEntity {

	private static final long serialVersionUID = 1L;

	private Long studentId;
	
	private String studentName;
	
	private Long userId;

	private Long specialtyId;

	private String specialtyName;

	private Integer campusId;
	
	private Integer termId;

	private String rightTime;
	
	private String classIds;

	private Boolean isEnable;


}
